cpp/modmesh/transform/fourier.hpp

```diff
@@ -72,24 +50,71 @@ void fft_radix_2(SimpleArray<T1<T2>> const & in, SimpleArray<T1<T2>> & out)
     }
 }
 
-template <template <typename> class T1, typename T2>
-void ifft(SimpleArray<T1<T2>> const & in, SimpleArray<T1<T2>> & out)
-{
-    size_t N = in.size();
-    SimpleArray<T1<T2>> in_conj{modmesh::small_vector<size_t>{N}, T1<T2>{0.0, 0.0}};
+} /* end namespace detail */
 
-    for (size_t i = 0; i < N; ++i)
+class Transform
```

Comment:
To prepare for the Python wrapper, encapsulate the Fourier transform interface in `Transform` class.

Reply:
Please don't use the term `Transform`, which will be more useful for geometry transform.  Please name it explicitly as `FourierTransform`.